test/functional/api/vim_spec.lua

```diff
@@ -399,6 +399,19 @@ describe('API', function()
       ]],
       }
     end)
+
+    it('errors properly when command too recursive', function()
+      exec_lua([[
+        _G.success = false
+        vim.api.nvim_create_user_command('Test', function()
+          vim.api.nvim_exec2('Test', {})
+          _G.success = true
+        end, {})
+      ]])
+      pcall_err(command, 'Test')
```

Comment:
It'd be great if the error message printed when stack overflows can be improved. (e.g. E169) The current one might look too messy? lots of repeated `Vim: Error executing Lua callback ....` string in a single line.

Reply:
This sounds quite complicated as the nesting can happen in any arbitrary structure.